processing/src/main/java/org/apache/druid/segment/serde/DictionaryEncodedStringIndexSupplier.java

```diff
@@ -98,21 +113,21 @@ public ColumnIndexCapabilities getIndexCapabilities()
     }
   }
 
-  private abstract static class BaseGenericIndexedDictionaryEncodedStringIndex
+  private abstract static class BaseGenericIndexedDictionaryEncodedIndex<T>
   {
     protected final BitmapFactory bitmapFactory;
-    protected final GenericIndexed<String> dictionary;
-    protected final GenericIndexed<ImmutableBitmap> bitmaps;
+    protected final Indexed<T> dictionary;
+    protected final Indexed<ImmutableBitmap> bitmaps;
 
-    protected BaseGenericIndexedDictionaryEncodedStringIndex(
+    protected BaseGenericIndexedDictionaryEncodedIndex(
         BitmapFactory bitmapFactory,
-        GenericIndexed<String> dictionary,
+        GenericIndexed<T> dictionary,
         GenericIndexed<ImmutableBitmap> bitmaps
     )
     {
       this.bitmapFactory = bitmapFactory;
-      this.dictionary = dictionary;
-      this.bitmaps = bitmaps;
+      this.dictionary = dictionary.singleThreaded();
```

Comment:
is a shame that this method loses the more specific type, but is still probably better